src/shared_utilities/implicit_tendencies.jl

```diff
@@ -1,5 +1,4 @@
-export make_tendency_jacobian,
-    make_update_jacobian, AbstractTridiagonalW, ∂tendencyBC∂Y
+export make_tendency_jacobian, make_update_jacobian, set_dfluxBCdY!
```

Comment:
I wonder if this should have a clearer name, or if we need this function at all. We can get rid of this function if we move the `update_aux!` and `update_boundary_fluxes!` function calls that `make_tendency_jacobian` does before it calls update jacobian into update_jacobian.

Reply:
Good point! I think it makes more sense to get rid of `make_tendency_jacobian`, and just have `make_update_jacobian`

Reply:
I realized that we're updating the aux and boundary fluxes in `exp_tendency!`, `imp_tendency!`, and `update_jacobian!`. I think this is fine for now, but we might want to avoid recomputing these multiple times per timestep. Dennis mentioned that ClimaTimeSteppers has the capability to take in an update cache function, which will be called once per timestep (Atmos leverages this to call `set_precomputed_quantities!` as few times as possible)., so we could later on switch to using that as a performance improvement

Reply:
After removing `tendency_jacobian!`, we can't directly compare the jacobian entries in `test/shared_utilities/implicit_timestepping/richards_model.jl`, since `update_jacobian!` also calls `update_aux` and `boundary_fluxes`. I wonder if it's still worth keeping separate functions for cleaner structure and more granular testing